lesson_2/calculator.py

```diff
@@ -1,16 +1,44 @@
-# calculator program that will:
-    # Ask the user for two numbers.
-    # Ask the user for operation to perform: add, subtract, multiply or divide.
-    # Perform the calculation and display the results
+'''calculator program that will:
+    Ask the user for two numbers.
+    Ask the user for operation to perform: add, subtract, multiply or divide.
+    Perform the calculation and display the results'''
 
-print('Welcome to Calculator!')
+def prompt(message):
+    print(f"==> {message}")
 
-# no error for input/ will crash if input is wrong --> would add error if was function
-num1 = int(input('Enter a number: '))
-num2 = int(input('Enter another number: '))
+def invalid_input(str_num):
+    try:
+        int(str_num)
+    except ValueError:
+        return True
+    return False
 
-operation = input('What operation would you like to perform?\n'
-                  '1) Addition 2) Subtraction 3) Multiplication 4) Division: ')
+prompt('Welcome to Calculator!')
+
+# NUM1
+prompt('Enter a number!')
+num1 = input()
+while invalid_input(num1):
+    prompt("Error: That isn't a valid number, expecting an integer!")
+    num1 = input()
+num1 = int(num1) # ugly conversion to int, but following tut they did
+# -- def know i can make better
```

Comment:
nit: refrain from committing these comments to the code, instead add a comment to this PR. It's common practice for you to open a PR and then add comments explaining your decisions to reviewers. If you commit these comments it will just clutter this code. If someone wants to know why you made a decision they can just reference the PR in the future and see your comment.